File: Sosyolobi.Api/DTOs/Activities/CreateActivityRequest.cs
using Sosyolobi.Api.Enums;

namespace Sosyolobi.Api.DTOs.Activities;

public class CreateActivityRequest
{
    public Guid CategoryId { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public DateTime EventDate { get; set; }
    public int NeededPeopleCount { get; set; }
    public decimal? PricePerPerson { get; set; }
    public SkillLevel SkillLevel { get; set; } = SkillLevel.Any;
    public GenderPreference GenderPreference { get; set; } = GenderPreference.Any;
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public string AddressText { get; set; } = null!;
    public string? AddressDetailPrivate { get; set; }
}

File: sosyolobi-mobile/lib/features/activities/domain/create_activity_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../../core/domain/enum_converters.dart';
import '../../../core/domain/enums.dart';

part 'create_activity_request.freezed.dart';
part 'create_activity_request.g.dart';

/// Mirrors `Sosyolobi.Api/DTOs/Activities/CreateActivityRequest.cs` — the
/// 3-step wizard's submit payload (category → details → map pin-drop).
@freezed
abstract class CreateActivityRequest with _$CreateActivityRequest {
  const factory CreateActivityRequest({
    required String categoryId,
    required String title,
    String? description,
    required DateTime eventDate,
    required int neededPeopleCount,
    double? pricePerPerson,
    @SkillLevelConverter() @Default(SkillLevel.any) SkillLevel skillLevel,
    @GenderPreferenceConverter() @Default(GenderPreference.any) GenderPreference genderPreference,
    required double latitude,
    required double longitude,
    required String addressText,
    String? addressDetailPrivate,
  }) = _CreateActivityRequest;

  factory CreateActivityRequest.fromJson(Map<String, dynamic> json) => _$CreateActivityRequestFromJson(json);
}
